### PR TITLE
Add Siglis zigfred uno smart switch/relay/dimmer combination

### PR DESCRIPTION
**Pull request by Atokulus**
This adds Siglis zigfred uno smart switch/relay/dimmer combination.

Working out of the box:

- Relay control
- Dimmer control including brightness
- Front LED color (XY), brightness and on/off
- Requiring custom device handler for button events

Support has been added to Zigbee2MQTT just recently, see [devices/siglis.js](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/siglis.js).

Many thanks go to @agners

### CODE REVIEW

**Comment by dontinelli** [comment]
Wow, very quick fix. Hopefully this PR will be merged quickly into base branch.

**Comment by agners** [comment]
@Atokulus thanks for this PR! As you noted in #1292 this requires firmware 28, at this point I haven't (yet) received firmware 28 on the app yet. I will report back once I have the firmware upgrade installed!

File: zhaquirks/siglis/zigfred.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ) -> None:
+        """Handle cluster specific commands."""
+        if hdr.command_id == ZIGFRED_CLUSTER_COMMAND_BUTTON_EVENT:
+            self._process_button_event(args[0])
+
+    # def _update_attribute(self, attrid, value):
```
Comment (dmulcahey) [inline zhaquirks/siglis/zigfred.py:112]: Can you remove the commented out code please?

**Review by dmulcahey — APPROVED** [review]
I am going to merge this so it makes the release. Please open a subsequent PR to remove the commented out code. Thanks for the contribution!

File: zhaquirks/siglis/zigfred.py (hunk 1 above)
Comment (Atokulus) [reply zhaquirks/siglis/zigfred.py:112]: Thanks @dmulcahey
I just did a pull request with the commented code removed.

**Comment by agners** [comment]
@Atokulus just tested this device handler with the firmware 28: Buttons work flawless! I configured them to do "nothing" in the zigfred app, and I can see all events for each button individually now on HA side (short press/long press/release). :+1: 

Light control for the relay and dimmer works fine too.

The only thing which does not seem to work quite right is the front LED color change: I get the color wheel in HA, I can select a color. Selecting a color also changes the color on the front panel. However, HA seems not to get the update that the color indeed changed (it jumps back/stays at the original color).  It seems that ZHA relies on the feedback from the lamp, which seems not to be received in zigfreds case.

**Comment by dontinelli** [comment]
Good to hear, that it works.
Unfortunately the firmware update has not yet showed up in my app, even after deleting the cache and the app memory (is there any way to force a look up of the current firmware)?
@agners: have you just downloaded the new files into the quirk directory to update HA?

**Comment by agners** [comment]
@dontinelli the firmware is not yet available via regular app. I have a beta version of the app which allows to install beta firmware.

I did copy it manually to the right place in my dev environment (/usr/local/lib/python3.9/site-packages/zhaquirks/siglis/zigfred.py).

But the current beta of HA core 2022.3 already updated to the latest zha-device-handlers which contains the quirk.

**Comment by dontinelli** [comment]
Thnx for the explenation. As the release notes of the new firmware are already on the homepage since Wednesday (with today as release day) I thougt to have a problem on my side.

**Pull request merged**